Refactor to use mustache templates

ekohl opened the pull request:
This makes it easier to customize the generated spec file. It also includes other fixes from https://github.com/dLobatog/npm2rpm/pull/34.

ekohl committed:
Refactor to use mustache templates

This makes it easier to customize the generated spec file. It also
includes other fixes

ekohl commented on template.mustache:
```diff
@@ -0,0 +1,91 @@
+for tgz in %{sources}; do
+  echo $tgz | grep -q registry.npmjs.org || npm cache add --cache %{npm_cache_dir} $tgz
+done
+%setup -T -q -a {{SOURCES.length-1}} -D -n %{npm_cache_dir}
```
note to self: this doesn't work

dLobatog reviewed (commented):
@ekohl https://github.com/dLobatog/npm2rpm/pull/34/ changes the default template, if this is changing again should I review this one directly instead..?

ekohl commented:
It includes what's in #34 but I think that merging #34 first is better. I can easily rebase on it and fix the remaining issues in this PR.

ekohl committed:
Use explicit dependencies for single style packages

nodejs_find_provides_and_requires is noop in EPEL so this results in
broken packages.

ekohl commented on template.mustache:
```diff
@@ -0,0 +1,93 @@
+for tgz in %{sources}; do
+  echo $tgz | grep -q registry.npmjs.org || npm cache add --cache %{npm_cache_dir} $tgz
+done
+%setup -T -q -a {{PROVIDES.length}} -D -n %{npm_cache_dir}
```
I'm not 100% sure about the PROVIDES.length here or if there'll be an off-by-one error.

ekohl reviewed (commented):
I think this should be mostly OK to go. Anyone mind having a look at the provides comment?

ekohl commented:
Superseded by https://github.com/dLobatog/npm2rpm/pull/36
